### ISSUE TITLE
Grid and TreeList Date Filter Input Renders with "undefined" Placeholder in the Grid Filter Menu

### ISSUE THREAD

**Issue by yanmariomenev** [telerik/kendo-angular#4670]
### Description

When opening the filter menu for a date-type column in the Kendo UI Grid, the date/time input sometimes renders with placeholder="undefined".

This causes accessibility issues:

Screen readers (e.g., NVDA) announce "undefined" as the field’s name or placeholder, confusing users.
Accessibility tools (e.g., WAVE) flag this as a failure.

### Steps To Reproduce

Open the example from our [documentation](https://www.telerik.com/kendo-angular-ui/components/grid/filtering/filter-menu#angular-grid-filter-menu) and inspect the DatePicker inside the date filter component.

### Screenshots or video

<img width="1250" height="597" alt="Image" src="https://github.com/user-attachments/assets/8531ac85-90a7-498b-a010-e73a0170cba9" />

### Actual Behavior

.

### Expected Behavior

The placeholder should never be set to "undefined". If no placeholder is provided, it should be omitted or use a localized example format (e.g., "dd.mm.yyyy").

### Browser

Chrome

### Browser version

latest

### OS type

MacOS
